### ISSUE TITLE
rewrite the help output

### ISSUE THREAD

**Issue by bmesuere** [unipept/unipept#257]
`@tvwillem` : let me know if you have checked this, then I will review it

_[Original issue](https://github.ugent.be/unipept/unipept/issues/257) by ghost on Mon Apr 28 2014 at 18:33._
_Closed by @bmesuere on Wed May 20 2015 at 18:36._

**Comment by pdawyndt** [comment]
`@tvwillem` @bmesuere Here's a first attempt to have a consistent list of options and a clean help file for each of the command of the Unipept API command line interface. Before implementing the changes, all documentation should be critically reviewed. Here are some points of attention and some points for further discussion.
- commands and their options already have been modified according to some open issues
- it may be relevant to include a list of available information fields to the commands that have tabular output
- I would remove the aliases for the `unipept` subcommands, since they do not seem to add value _Peter_
- I would remove the `-x`/`--xml` option from the `unipept pept2prot` command, as this functionality would fit better with the `uniprot` command _Peter_
- I would remove the `-x`/`--xml` option from the `unipept pept2taxa` and `unipept pept2lca` commands, as this functionality would fit better with a wrapper command for the NCBI Taxonomy web services (e.g. `ncbi taxonomy`), analogous to the `uniprot` command _Peter_


_[Original comment](https://github.ugent.be/unipept/unipept/issues/257#issuecomment-10038) by @pdawyndt on Wed Jun 11 2014 at 22:19._

**Comment by pdawyndt** [comment]
**unipept**

```
NAME
    unipept - Command line interface to Unipept web services.

USAGE
    unipept subcommand [option]... [argument]...

DESCRIPTION
    The unipept subcommands are command line wrappers around the Unipept web services.

    Subcommands that start with pept expect a list of tryptic peptides as input. Subcommands that start with tax expect a list of NCBI Taxonomy Identifiers as input. Input is passed

        ° as separate command line arguments
        ° in one or more text files that are passed as an argument to the -i option
        ° to standard input

    The command will give priority to the first way the input is passed, in the order as listed above. Text files and standard input should have one tryptic peptide or one NCBI Taxonomy Identifier per line. 

    The unipept pept2prot subcommand yields Uniprot records as output. All other unipept subcommands yield NCBI Taxonomy records as output.

SUBCOMMANDS
    pept2prot     Fetch Uniprot records that match tryptic peptides.
    pept2taxa     Fetch taxa of Uniprot records that match tryptic peptides.
    pept2lca      Fetch taxonomic lowest common ancestor of Uniprot records that match tryptic peptides.
    taxa2lca      Compute taxonomic lowest common ancestor for given list of taxa.
    taxonomy      Fetch taxonomic information from Unipept Taxonomy.
    config        Configure default values for command line options.
    help          Display this help message and exit.

OPTIONS
    General Options
        -f output-format 
        --format=output-format
            Define output format as output-format (options: csv (default), json, xml).

        -h
        --help
            Display this help message and exit.

        -i file
        --input=input-file
            Read input from input-file. Multiple -i (or --input) options may be used.

        -o file 
        --output=output-file
            Write output to output-file. By default output is written to standard output.

        -q
        --quiet
            Turn off server side messages.

        -v
        --version
            Display version and exit.

        --host=hostname
            Set hostname of machine providing Unipept web services to hostname.

        --log=log-file
            Log all messages that are normally reported to standard error to log-file (default: ~/.unipept).
```


_[Original comment](https://github.ugent.be/unipept/unipept/issues/257#issuecomment-10039) by @pdawyndt on Wed Jun 11 2014 at 22:19._

**Comment by pdawyndt** [comment]
**unipept pept2prot**

```
NAME
    unipept pept2prot - Fetch Uniprot records that match tryptic peptides.

    aliases: p

USAGE
    unipept pept2prot [option]... [argument]...

DESCRIPTION
    For each tryptic peptide the unipept pept2prot command retrieves from Unipept all Uniprot records whose protein sequence contains an exact matches to the tryptic peptide. The command expects a list of tryptic peptides that are passed

        ° as separate command line arguments
        ° in one or more text files that are passed as an argument to the -i option
        ° to standard input

    The command will give priority to the first way tryptic peptides are passed, in the order as listed above. Text files and standard input should have one tryptic peptide per line.

    The unipept pept2prot subcommand yields Uniprot records as output. 

OPTIONS
    Command-specific Options
        -a 
        --all
            Report all information fields of Uniprot records available in Unipept. Note that retrieving all available information may have an additional performance penalty.

        -e
        --equate
            Matching algorithm equates isoleucine (I) and leucine (L).

        -s field1,field2,...
        --select=field1,field2,...
            Report selected information fields of Uniprot records available in Unipept. Selected fields are passed as a command separated list of field names. Multiple -s (or --select) options may be used.

        -x download-directory
        --xml=download-directory
            Download XML formatted Uniprot records from the Uniprot web services into download-directory. 

    General Options
        -f output-format 
        --format=output-format
            Define output format as output-format (options: csv (default), json, xml).

        -h
        --help
            Display this help message and exit.

        -i file
        --input=input-file
            Read input from input-file. Multiple -i (or --input) options may be used.

        -o file 
        --output=output-file
            Write output to output-file. By default output is written to standard output.

        -q
        --quiet
            Turn off server side messages.

        -v
        --version
            Display version and exit.

        --host=hostname
            Set hostname of machine providing Unipept web services to hostname.

        --log=log-file
            Log all messages that are normally reported to standard error to log-file (default: ~/.unipept).
```


_[Original comment](https://github.ugent.be/unipept/unipept/issues/257#issuecomment-10040) by @pdawyndt on Wed Jun 11 2014 at 22:20._

**Comment by pdawyndt** [comment]
**unipept pept2taxa**

```
NAME
    unipept pept2taxa - Fetch taxa of Uniprot records that match tryptic peptides.

    aliases: s

USAGE
    unipept pept2taxa [option]... [argument]...

DESCRIPTION
    For each tryptic peptide the unipept pept2taxa command retrieves from Unipept the set of taxa from all Uniprot records whose protein sequence contains an exact matches to the tryptic peptide. The command expects a list of tryptic peptides that are passed

        ° as separate command line arguments
        ° in one or more text files that are passed as an argument to the -i option
        ° to standard input

    The command will give priority to the first way tryptic peptides are passed, in the order as listed above. Text files and standard input should have one tryptic peptide per line.

    The unipept pept2taxa subcommand yields NCBI Taxonomy records as output.

OPTIONS
    Command-specific Options
        -a 
        --all
            Report all information fields of NCBI Taxonomy records available in Unipept. Note that retrieving all available information may have an additional performance penalty.

        -e
        --equate
            Matching algorithm equates isoleucine (I) and leucine (L).

        -s field1,field2,...
        --select=field1,field2,...
            Report selected information fields of NCBI Taxonomy records available in Unipept. Selected fields are passed as a command separated list of field names. Multiple -s (or --select) options may be used.

        -x download-directory
        --xml=download-directory
            Download XML formatted NCBI Taxonomy records from the NCBI Taxonomy web services into download-directory. 

    General Options
        -f output-format 
        --format=output-format
            Define output format as output-format (options: csv (default), json, xml).

        -h
        --help
            Display this help message and exit.

        -i file
        --input=input-file
            Read input from input-file. Multiple -i (or --input) options may be used.

        -o file 
        --output=output-file
            Write output to output-file. By default output is written to standard output.

        -q
        --quiet
            Turn off server side messages.

        -v
        --version
            Display version and exit.

        --host=hostname
            Set hostname of machine providing Unipept web services to hostname.

        --log=log-file
            Log all messages that are normally reported to standard error to log-file (default: ~/.unipept).
```


_[Original comment](https://github.ugent.be/unipept/unipept/issues/257#issuecomment-10041) by @pdawyndt on Wed Jun 11 2014 at 22:20._

**Comment by pdawyndt** [comment]
**unipept pept2lca**

```
NAME
    unipept pept2lca - Fetch taxonomic lowest common ancestor of Uniprot records that match tryptic peptides.

    aliases: l

USAGE
    unipept pept2lca [option]... [argument]...

DESCRIPTION
    For each tryptic peptide the unipept pept2lca command retrieves from Unipept the lowest common ancestor of the set of taxa from all Uniprot records whose protein sequence contains an exact matches to the tryptic peptide. The lowest common ancestor is based on the topology of the Unipept Taxonomy -- a cleaned up version of the NCBI Taxonomy -- and is itself a record from the NCBI Taxonomy. The command expects a list of tryptic peptides that are passed

        ° as separate command line arguments
        ° in one or more text files that are passed as an argument to the -i option
        ° to standard input

    The command will give priority to the first way tryptic peptides are passed, in the order as listed above. Text files and standard input should have one tryptic peptide per line. 

    The unipept pept2lca subcommand yields an NCBI Taxonomy record as output.

OPTIONS
    Command-specific Options
        -a 
        --all
            Report all information fields of NCBI Taxonomy records available in Unipept. Note that retrieving all available information may have an additional performance penalty.

        -e
        --equate
            Matching algorithm equates isoleucine (I) and leucine (L).

        -s field1,field2,...
        --select=field1,field2,...
            Report selected information fields of Uniprot records available in Unipept. Selected fields are passed as a command separated list of field names. Multiple -s (or --select) options may be used.

        -x download-directory
        --xml=download-directory
            Download XML formatted NCBI Taxonomy records from the NCBI Taxonomy web services into download-directory. 

    General Options
        -f output-format 
        --format=output-format
            Define output format as output-format (options: csv (default), json, xml).

        -h
        --help
            Display this help message and exit.

        -i file
        --input=input-file
            Read input from input-file. Multiple -i (or --input) options may be used.

        -o file 
        --output=output-file
            Write output to output-file. By default output is written to standard output.

        -q
        --quiet
            Turn off server side messages.

        -v
        --version
            Display version and exit.

        --host=hostname
            Set hostname of machine providing Unipept web services to hostname.

        --log=log-file
            Log all messages that are normally reported to standard error to log-file (default: ~/.unipept).
```


_[Original comment](https://github.ugent.be/unipept/unipept/issues/257#issuecomment-10042) by @pdawyndt on Wed Jun 11 2014 at 22:20._

**Comment by pdawyndt** [comment]
**unipept taxa2lca**

```
NAME
    unipept taxa2lca - Compute taxonomic lowest common ancestor for given list of taxa.

    aliases: t

USAGE
    unipept taxa2lca [option]... [argument]...

DESCRIPTION
    The unipept taxa2lca command computes the lowest common ancestor of a given list of NCBI Taxonomy Identifiers. The lowest common ancestor is based on the topology of the Unipept Taxonomy -- a cleaned up version of the NCBI Taxonomy -- and is itself a record from the NCBI Taxonomy. The command expects a list of NCBI Taxonomy Identifiers that are passed

        ° as separate command line arguments
        ° in one or more text files that are passed as an argument to the -i option
        ° to standard input

    The command will give priority to the first way NCBI Taxonomy Identifiers are passed, in the order as listed above. Text files and standard input should have one NCBI Taxonomy Identifier per line. 

    The unipept taxonomy subcommand yields NCBI Taxonomy records as output.

OPTIONS
    Command-specific Options
        -a 
        --all
            Report all information fields of NCBI Taxonomy records available in Unipept. Note that retrieving all available information may have an additional performance penalty.

        -s field1,field2,...
        --select=field1,field2,...
            Report selected information fields of NCBI Taxonomy records available in Unipept. Selected fields are passed as a command separated list of field names. Multiple -s (or --select) options may be used.

        -x download-directory
        --xml=download-directory
            Download XML formatted NCBI Taxonomy records from the NCBI Taxonomy web services into download-directory. 

    General Options
        -f output-format 
        --format=output-format
            Define output format as output-format (options: csv (default), json, xml).

        -h
        --help
            Display this help message and exit.

        -i file
        --input=input-file
            Read input from input-file. Multiple -i (or --input) options may be used.

        -o file 
        --output=output-file
            Write output to output-file. By default output is written to standard output.

        -q
        --quiet
            Turn off server side messages.

        -v
        --version
            Display version and exit.

        --host=hostname
            Set hostname of machine providing Unipept web services to hostname.

        --log=log-file
            Log all messages that are normally reported to standard error to log-file (default: ~/.unipept).
```


_[Original comment](https://github.ugent.be/unipept/unipept/issues/257#issuecomment-10043) by @pdawyndt on Wed Jun 11 2014 at 22:20._

**Comment by pdawyndt** [comment]
**unipept taxonomy**

```
NAME
    unipept taxonomy - Fetch taxonomic information from Unipept Taxonomy.

    aliases: tax

USAGE
    unipept taxonomy [options]

DESCRIPTION
    The unipept taxonomy command yields information from the Unipept Taxonomy records for a given list of NCBI Taxonomy Identifiers. The Unipept Taxonomy is a cleaned up version of the NCBI Taxonomy, and its records are also records of the NCBI Taxonomy. The command expects a list of NCBI Taxonomy Identifiers that are passed

        ° as separate command line arguments
        ° in one or more text files that are passed as an argument to the -i option
        ° to standard input

    The command will give priority to the first way NCBI Taxonomy Identifiers are passed, in the order as listed above. Text files and standard input should have one NCBI Taxonomy Identifier per line. 

    The unipept taxonomy subcommand yields NCBI Taxonomy records as output.

OPTIONS
    Command-specific Options
        -a 
        --all
            Report all information fields of NCBI Taxonomy records available in Unipept. Note that retrieving all available information may have an additional performance penalty.

        -s field1,field2,...
        --select=field1,field2,...
            Report selected information fields of NCBI Taxonomy records available in Unipept. Selected fields are passed as a command separated list of field names. Multiple -s (or --select) options may be used.

    General Options
        -f output-format 
        --format=output-format
            Define output format as output-format (options: csv (default), json, xml).

        -h
        --help
            Display this help message and exit.

        -i file
        --input=input-file
            Read input from input-file. Multiple -i (or --input) options may be used.

        -o file 
        --output=output-file
            Write output to output-file. By default output is written to standard output.

        -q
        --quiet
            Turn off server side messages.

        -v
        --version
            Display version and exit.

        --host=hostname
            Set hostname of machine providing Unipept web services to hostname.

        --log=log-file
            Log all messages that are normally reported to standard error to log-file (default: ~/.unipept).
```


_[Original comment](https://github.ugent.be/unipept/unipept/issues/257#issuecomment-10044) by @pdawyndt on Wed Jun 11 2014 at 22:21._

**Comment by pdawyndt** [comment]
**uniprot**

```
NAME
    uniprot - Command line interface to Uniprot web services.

USAGE
    uniprot [subcommand] [option]... [argument]...

DESCRIPTION
    The uniprot command is a command line wrapper around the Uniprot web services. The command expects a list of Uniprot Accession Numbers that are passed

        ° as separate command line arguments
        ° in one or more text files that are passed as an argument to the -i option
        ° to standard input

    The command will give priority to the first way Uniprot Accession Numbers are passed, in the order as listed above. Text files and standard input should have one Uniprot Accession Number per line.

    The uniprot command yields Uniprot records as output. 

SUBCOMMANDS
    help          Display this help message and exit.

OPTIONS
    -a 
    --all
        Report all information fields of Uniprot records. Note that retrieving all available information may have an additional performance penalty.

    -f output-format 
    --format=output-format
        Define output format as output-format (options: csv (default), fasta, gff, rdf, text, xml).

    -h
    --help
        Display this help message and exit.

    -i file
    --input=input-file
        Read input from input-file. Multiple -i (or --input) options may be used.

    -o file 
    --output=output-file
        Write output to output-file. By default output is written to standard output.

    -v
    --version
        Display version and exit.

    --log=log-file
        Log all messages that are normally reported to standard error to log-file (default: ~/.uniprot).
```


_[Original comment](https://github.ugent.be/unipept/unipept/issues/257#issuecomment-10045) by @pdawyndt on Wed Jun 11 2014 at 22:21._

**Comment by pdawyndt** [comment]
**prot2pept**

```
NAME
    prot2pept - Split protein sequences into peptides.

USAGE
    prot2pept [subcommand] [option]... [argument]...

DESCRIPTION
    The prot2pept command splits each protein sequence into a list of peptides according to a given cleavage-pattern. The command expects a list of protein sequences that are passed

        ° as separate command line arguments
        ° in one or more text files that are passed as an argument to the -i option
        ° to standard input

    The command will give priority to the first way protein sequences are passed, in the order as listed above. Text files and standard input should have either one protein sequence per line or contain a FASTA formatted list of protein sequences. FASTA headers are preserved in the output, so that peptides can be bundled per protein sequence.

SUBCOMMANDS
    help          Display this help message and exit.

OPTIONS
    -h
    --help
        Display this help message and exit.

    -i file
    --input=input-file
        Read input from input-file. Multiple -i (or --input) options may be used.

    -o file 
    --output=output-file
        Write output to output-file. By default output is written to standard output.

    -p cleavage-pattern
    --pattern=cleavage-pattern    
        Specify cleavage-pattern (regex) as the pattern after which the next peptide will be cleaved (default: ([KR])([^P]) for tryptic peptides).

    -v
    --version
        Display version and exit.

    --log=log-file
        Log all messages that are normally reported to standard error to log-file (default: ~/.uniprot).
```


_[Original comment](https://github.ugent.be/unipept/unipept/issues/257#issuecomment-10046) by @pdawyndt on Wed Jun 11 2014 at 22:21._

**Comment by pdawyndt** [comment]
**peptfilter**

```
NAME
    peptfilter - Filter peptides based on specific criteria.

USAGE
    peptfilter [subcommand] [option]... [argument]...

DESCRIPTION
    The peptfilter command filters a list of peptides according to specific criteria. The command expects a list of peptides that are passed

        ° as separate command line arguments
        ° in one or more text files that are passed as an argument to the -i option
        ° to standard input

    The command will give priority to the first way peptides are passed, in the order as listed above. Text files and standard input should have one peptide per line. FASTA headers are preserved in the output, so that peptides remain bundled.

SUBCOMMANDS
    help          Display this help message and exit.

OPTIONS
    -l residues
    --lacks=residues
        Only retain tryptic peptides that lack all amino acids from the string of residues.

    -c
    --contains
        Only retain tryptic peptides that contain all amino acids from the string of residues.

    -h
    --help
        Display this help message and exit.

    -i file
    --input=input-file
        Read input from input-file. Multiple -i (or --input) options may be used.

    -L min
    --minlen=min
        Only retain tryptic peptides that have at least min (default: 5) amino acids.

    -o file 
    --output=output-file
        Write output to output-file. By default output is written to standard output.

    -u
    --unique
        Deduplicate list of tryptic peptides.

    -U max
    --maxlen=max
        Only retain tryptic peptides that have at most max (default: 50) amino acids.

    -v
    --version
        Display version and exit.

    --log=log-file
        Log all messages that are normally reported to standard error to log-file (default: ~/.uniprot).
```


_[Original comment](https://github.ugent.be/unipept/unipept/issues/257#issuecomment-10047) by @pdawyndt on Wed Jun 11 2014 at 22:21._

**Comment by pdawyndt** [comment]
`@tvwillem` @bmesuere : Toon, Bart and I have reviewed the documentation of the command line tools today, and it seems they are now stable; please go ahead with implementing them


_[Original comment](https://github.ugent.be/unipept/unipept/issues/257#issuecomment-10058) by @pdawyndt on Thu Jun 12 2014 at 17:55._

**Comment by bmesuere** [comment]
The help output of all implemented functions was updated for version 0.7. There are a few features still on the to do list, so adding help for them isn't useful yet. This issue was linked in all issues of that to do list.


_[Original comment](https://github.ugent.be/unipept/unipept/issues/257#issuecomment-18119) by @bmesuere on Wed May 20 2015 at 18:36._
